apps/src/gamelab/spritelab/spriteUtils.js

```diff
@@ -122,90 +122,104 @@ export function addEvent(type, args, callback) {
 }
 
 function whenPressEvent(inputEvent, p5Inst) {
-  return {
-    shouldEventFire: p5Inst.keyWentDown(inputEvent.args.key),
-    extraArgs: {}
-  };
+  if (p5Inst.keyWentDown(inputEvent.args.key)) {
+    // Call callback with no extra args
+    return [{}];
+  } else {
+    // Don't call callback
+    return [];
+  }
 }
 
 function whilePressEvent(inputEvent, p5Inst) {
-  return {shouldEventFire: p5Inst.keyDown(inputEvent.args.key), extraArgs: {}};
+  if (p5Inst.keyDown(inputEvent.args.key)) {
+    // Call callback with no extra args
+    return [{}];
+  } else {
+    // Don't call callback
+    return [];
+  }
 }
 
 function whenTouchEvent(inputEvent) {
```

Comment:
I'm not at all convinced that this is the right approach for this function... Curious to hear people's thoughts and/or talk through it and see if there's a cleaner way!